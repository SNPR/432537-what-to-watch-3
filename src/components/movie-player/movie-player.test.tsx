import * as React from "react";
import renderer from "react-test-renderer";
import MoviePlayer from "./movie-player";

const movie = {
  name: `Movie name`,
  posterUrl: `https://poster-url.com`,
  bigPosterUrl: `https://image-url.com/1.jpg`,
  trailerUrl: `https://upload.wikimedia.org/wikipedia/commons/d/d2/Frankenstein_trailer_%281931%29.webm`,
  director: `Director Name`,
  starring: [`Actor 1`, `Actor 2`, `Actor 3`],
  runTime: `1h 00m`,
  genre: `Movie Genre`,
  releaseYear: 2000,
  rating: 8.9,
  votes: 4235,
  description: `Movie description`,
  reviews: [
    {
      rating: 9,
      date: `November 10, 2019`,
      author: `Dmitriy`,
      text: `Review text`
    }
  ]
};

it(`Should render MoviePlayer component`, () => {
  const tree = renderer
    .create(<MoviePlayer movie={movie} muted={true} autoPlay={true} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
