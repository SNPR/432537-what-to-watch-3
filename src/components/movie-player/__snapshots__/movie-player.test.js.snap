// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MoviePlayer component 1`] = `
<video
  controls={true}
  muted={true}
  onClick={[Function]}
  poster="https://poster-url.com"
  width="100%"
>
  <source
    src="https://upload.wikimedia.org/wikipedia/commons/d/d2/Frankenstein_trailer_%281931%29.webm"
  />
</video>
`;
